ten-simple-rules-dockerfiles.Rmd

```diff
@@ -292,7 +316,7 @@ A recommended ordering based on these considerations is as follows.
 4. labels
 5. `RUN`/`ENTRYPOINT`
 
-Finally, as a supplement to content inside the `Dockerfile`, it is good practice to also write a section in a `README` alongside the `Dockerfile` for exactly how to build, run, and otherwise interact with the container. 
+Finally, as a supplement to content inside the `Dockerfile`, it is good practice to also write a section in a `README` alongside the `Dockerfile` for exactly how to build, run, and otherwise interact with the container.  [SJE: really?  Doesn't that mean you end up writing the instructions twice -- once in the Dockerfile as a comment, and once in the README?  Surely the README could just say:  "Please read the end of the Dockerfile for instructions on how to run the docker image."]
```

Comment:
Probably the redundancy is good, although problematic for the points you raise. Some users will read the Dockerfile, and others the README, and sometimes the Dockerfile is separated from the README. It's messy at best.

Reply:
I think referencing the instructions from the README is a nice idea. It also forces readers to take a look at the Dockerfile and know that it is there. OTOH it's a matter of taste, so maybe we just point out options and risks?

> Finally, as a supplement to content inside the `Dockerfile`, it is good practice to also write a section in a `README` about how to use the container. This section may duplicate the documentation from the end of the `Dockerfile` at the risk of diverging and inconsistency, or simply point the reader to the respective lines in the `Dockerfile`.